torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -279,6 +279,17 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                     node.replace_all_uses_with(guard_scalar(val))
                     graph.erase_node(node)
 
+    for node in graph.nodes:
+        # Need to do one more run because metas can now be inconsistent.
+        # eg. previously eg(_local_scalar_dense, inf) would have meta['val'] = False
+        # but after the previously specialization, this val is now True.
+        if (
+            callable(node.target)
+            and len(node.args) > 1
+            and all(isinstance(a, float) for a in node.args)
+        ):
+            node.meta["val"] = node.target(*node.args, **node.kwargs)
```

Comment:
This seems bad. Concretely, if a float input is infinity (zf0 == inf), we should refuse to generate a symbolic variable for it from the getgo, because we will specialize on this in ways that are unsound (like the meta['val'] = False you saw here). I think we already have this treatment for nan so you just have to extend it for inf/-inf. 

Relevant code:

```
            with no_dispatch():
                value = t.item()
            if not math.isnan(value):
                # Peephole strip out unnecessary torch.as_tensor(x).item()
                if isinstance(source, FloatTensorSource):
                    item_source = source.base
                else:
                    item_source = CallMethodItemSource(source)
                symbol = shape_env.create_unspecified_symbol(
                    value,
                    source=item_source,
                    dynamic_dim=DimDynamic.DYNAMIC,
                )
                # NB: reusing item_memo here ensures that we invalidate on
                # mutation
                if t.dtype == torch.int64:
                    out.item_memo = shape_env.create_symintnode(
                        symbol, 
                        hint=value,
                        source=item_source,
                    )
                elif t.dtype == torch.float64:
                    out.item_memo = shape_env.create_symfloatnode(
                        symbol,
                        hint=value,
                        source=item_source,
                    )
```

in fake_tensor.py